lib/util/buffer-manager.ts

```diff
@@ -0,0 +1,37 @@
+/// <reference path='../../typings/tsd.d.ts' />
+
+export interface IBufferedData<T> {
+    data: T[];
+    overflow: number;
+}
+
+export class BufferManager<T> {
+
+    // maxBuffLength: Maximum numbers of data item per level of buffer
+    // maxHistoryDepth: Maximum numbers of preserved buffer history
+    constructor(maxBuffLength: number, maxHistoryDepth: number) {
```

Comment:
Let's switch these two parameters around and have the maxBuffLength be optional with a predefined default.